src/analytics/internal/sendToServer/SendToServer.js

```diff
@@ -0,0 +1,127 @@
+import { idleManager } from 'src/util/idle'
+import { SHOULD_TRACK_STORAGE_KEY as SHOULD_TRACK } from 'src/options/privacy/constants'
+import { USER_ID, generateTokenIfNot } from 'src/util/generate-token'
+import { installTimeStorageKey } from 'src/imports/background'
+
+class SendToServer {
+    static API_PATH = '/event'
+    static JSON_HEADER = {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+    }
+
+    /**
+     * Pool of requests that have been tracked, which will be periodically cleared and sent off in bulk.
+     */
+    _pool = new Set()
+
+    /**
+     * @param {Object} args
+     * @param {string} args.url Full URL of the analytics server host.
+     */
+    constructor({ url }) {
+        this._host = url + SendToServer.API_PATH
+
+        // Schedule sending of network req when user is idle or locks screen
+        idleManager.scheduleIdleCbs({
+            onIdle: this._sendBulkReq,
+            onLocked: this._sendBulkReq,
+        })
+    }
+
+    /**
+     *
+     * @param {params} params if params is there, then send request directly without wait, otherwise send pool request
+     */
+    async _serializePoolReqs(param = undefined) {
```

Comment:
This class looks a lot better :) one major remaining thing is this ambiguous `param` flag. I would prefer this method to take two params: `userId` and `events` array and simply shape the data object then call `JSON.stringify` on it to perform the serialization into JSON. Then both `_sendReq` and `_sendBulkReq` would call this method to create the `fetch` req's body, passing in the userId as first param but differing on the second param. I think then it could also just be a `static` method, as there won't be any access to `this`.

I also understand this will differ from the existing Analytics class, although I consider this to be a flaw in the existing class (`_serializePoolReqs` does not actually perform the serialization to JSON :S)

Also as a sidenote: there's no reason to assign `undefined` as a default param value in JS (if it's not specified when the function/method is called, it will be set to `undefined`).